code/unclassified/myk.md.txt

```diff
@@ -0,0 +1,78 @@
+// A hashmap based C++ program to clone a binary tree with random pointers
+#include<iostream>
+#include<map>
+using namespace std;
+
+/* A binary tree node has data, pointer to left child, a pointer to right
+child and a pointer to random node*/
+struct Node
+{
+	int key;
+	struct Node* left, *right, *random;
+};
+
+/* Helper function that allocates a new Node with the
+given data and NULL left, right and random pointers. */
+Node* newNode(int key)
+{
+	Node* temp = new Node;
+	temp->key = key;
+	temp->random = temp->right = temp->left = NULL;
+	return (temp);
+}
+
+/* Given a binary tree, print its Nodes in inorder*/
+void printInorder(Node* node)
+{
+	if (node == NULL)
+		return;
+
+	/* First recur on left sutree */
+	printInorder(node->left);
+
+	/* then print data of Node and its random */
+	cout << "[" << node->key << " ";
+	if (node->random == NULL)
+		cout << "NULL], ";
+	else
+		cout << node->random->key << "], ";
+
+	/* now recur on right subtree */
+	printInorder(node->right);
+}
+
+// This function creates clone by copying key and left and right pointers
+// This function also stores mapping from given tree node to clone.
+Node* copyLeftRightNode(Node* treeNode, map<Node *, Node *> *mymap)

```

Comment:
There should be no reason for a pointer to a map, in all the places you use one.